### PR TITLE
(0.30.0) PowerShell and Exec tasks should run in every mode

### PR DESCRIPTION
**Pull request by splatteredbits**
Currently, the PowerShell and Exec tasks can't be used to do things in clean and initialize modes. :( This pull request updates those tasks to run in all modes.

### CODE REVIEW

File: Test/Invoke-WhiskeyPowerShell.Tests.ps1 [hunk 1]
```diff
@@ -372,3 +386,23 @@ param(
     WhenTheTaskRuns -WithArgument @{ }
     ThenTheTaskPasses
 }
+
+Describe 'PowerShell.when run in Clean mode' {
+    GivenAScript @'
+New-Item -Path 'clean' -itemType 'File'
+'@
+    WhenTheTaskRuns -InCleanMode
+    It ('should run the script') {
```
Comment (jlariono) [inline Test/Invoke-WhiskeyPowerShell.Tests.ps1]: Nit: There is already a `ThenTheScriptRan` function that can test this.

File: Test/Invoke-WhiskeyPowerShell.Tests.ps1 [hunk 2]
```diff
@@ -372,3 +386,23 @@ param(
...
+        ThenFile -Path 'clean'
+    }
+}
+
+Describe 'PowerShell.when run in Initialize mode' {
+    GivenAScript @'
+New-Item -Path 'init' -itemType 'File'
+'@
+    WhenTheTaskRuns -InInitMode
+    It ('should run the script') {
```
Comment (jlariono) [inline Test/Invoke-WhiskeyPowerShell.Tests.ps1]: Nit: There is already a `ThenTheScriptRan` function that can test this.

File: Whiskey/Tasks/Invoke-WhiskeyPowerShell.ps1 [hunk 3]
```diff
@@ -1,7 +1,66 @@
...
+        param(
+            [object]
+            $TaskContext
+        )
+
+    This is *not* recommended.
+
+    ## Properties
+    * **Path** (mandatory): the paths to the PowerShell scripts to run. Paths must be relative to the  whiskey.yml file. Script arguments are not supported.
+    * **WorkingDirectory**: the directory from which the scripts should be run. This path can be absolute. The default is the  whiskey.yml file's directory.
```
Comment (jlariono) [inline Whiskey/Tasks/Invoke-WhiskeyPowerShell.ps1]: Isn't `WorkingDirectory` a global property now? Are there certain global properties we always include in task docs?

File: Whiskey/Tasks/Invoke-WhiskeyPowerShell.ps1 [hunk 4]
```diff
@@ -1,7 +1,66 @@
 
 function Invoke-WhiskeyPowerShell
 {
-    [Whiskey.Task("PowerShell")]
+    <#
+    .SYNOPSIS
+    Executes PowerShell tasks.
+
+    .DESCRIPTION
+    The PowerShell tasks runs PowerShell scripts. You specify the scripts to run via the `Path` property. Paths must be relative to the whiskey.yml file. Pass arguments to the scripts with the `Argument` property, which is a hash table of parameter names and values. PowerShell scripts are run in new, background processes.
```
Comment (jlariono) [inline Whiskey/Tasks/Invoke-WhiskeyPowerShell.ps1]: Typo: "PowerShell **tasks** runs" should be "PowerShell **task** runs".

File: Whiskey/Tasks/Invoke-WhiskeyExec.ps1 [hunk 5]
```diff
@@ -58,10 +60,33 @@ function Invoke-WhiskeyExec
...
+
+        BuildTasks:
+        - Exec:
+            ExceptDuring: Clean
+            Path: cmd.exe
+            Argument:
+            - /C
+            - init.bat
+
+    Demonstrates how to run an executable except when the build is cleaning. If you have am executable you want to use to initialize your build environment, it should run during the build and initialize modes. Set the `ExceptDuring` property to `Clean` to make that happen.
```
Comment (jlariono) [inline Whiskey/Tasks/Invoke-WhiskeyExec.ps1]: Typo: "have **am** executable" should be "have **an** executable".

**Comment by splatteredbits** [comment]
@jlariono Changes made. Please re-review.

**Pull request merged**